kernel/reduction.ml

```diff
@@ -267,8 +267,9 @@ let convert_constructors_gen cmp_instances cmp_cumul (mind, ind, cns) nargs u1 u
     if not (Int.equal num_cnstr_args nargs) then
       cmp_instances u1 u2 s
     else
-      let csts = get_cumulativity_constraints CONV cumi u1 u2 in
-      cmp_cumul csts s
+      (** By invariant, both constructors have a common supertype,
+          so they are convertible _at that type_. *)
```

Comment:
This should probably happen in the checker too.

Reply:
Right